No longer works with Prettier 3.6.0 and newer

Starting in [Prettier 3.6.0](https://prettier.io/blog/2025/06/23/3.6.0), the `prettier/internals/cli.mjs` script has been moved to `prettier/internals/legacy-cli.mjs`. Running `prettier-pnp` now fails:

```sh
$ prettier-pnp --pn sql --pn sh --pn java --pnp @prettier/plugin-xml --pnp prettier-plugin-ember-template-tag --write --config "~/.prettierrc.js" --ignore-path "~/.prettierignore"
----- Running prettier -----

node:internal/modules/esm/resolve:283
    throw new ERR_MODULE_NOT_FOUND(
          ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/wsl/.local/share/pnpm/global/5/.pnpm/prettier-pnp@0.3.0/node_modules/prettier/internal/cli.mjs' imported from /home/wsl/.local/share/pnpm/global/5/.pnpm/prettier-pnp@0.3.0/node_modules/prettier-pnp/dist/bin.js
    at finalizeResolution (node:internal/modules/esm/resolve:283:11)
    at moduleResolve (node:internal/modules/esm/resolve:952:10)
    at defaultResolve (node:internal/modules/esm/resolve:1188:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:708:12)
    at #cachedDefaultResolve (node:internal/modules/esm/loader:657:25)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:640:38)
    at ModuleLoader.getModuleJobForImport (node:internal/modules/esm/loader:264:38)
    at ModuleLoader.import (node:internal/modules/esm/loader:605:34)
    at defaultImportModuleDynamicallyForModule (node:internal/modules/esm/utils:221:31)
    at importModuleDynamicallyCallback (node:internal/modules/esm/utils:260:12) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///home/wsl/.local/share/pnpm/global/5/.pnpm/prettier-pnp@0.3.0/node_modules/prettier/internal/cli.mjs'
}

Node.js v20.19.6
```

It looks like the current version works with Prettier 3.5.3 at the latest.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -35,7 +35,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: [16, 24]
+        node: [18, 24]
         os: [ubuntu-latest, windows-latest]
 
     steps:

--- a/src/install-plugins.ts
+++ b/src/install-plugins.ts
@@ -124,6 +124,7 @@ export async function installPlugins(
       ],
       {
         cwd: PLUGIN_STORE_PATH,
+        shell: true,
         env: Object.assign({}, process.env, { NODE_ENV: 'production' }),
         stdio: extraArgs.quiet ? ['inherit', 'pipe', 'pipe'] : 'inherit',
       },

--- a/src/main.ts
+++ b/src/main.ts
@@ -19,7 +19,13 @@ export async function run(args: string[]) {
     }
   }
 
-  const prettierCli = await import('prettier/internal/cli.mjs')
+  let prettierCli: typeof import('prettier/internal/cli.mjs')
+  try {
+    // prettier 3.6+
+    prettierCli = await import('prettier/internal/legacy-cli.mjs')
+  } catch {
+    prettierCli = await import('prettier/internal/cli.mjs')
+  }
 
   const pnpPlugins: string[] = []
   pluginNames.forEach(pluginName => {

--- a/src/prettier-cli.d.ts
+++ b/src/prettier-cli.d.ts
@@ -1,3 +1,6 @@
 declare module 'prettier/internal/cli.mjs' {
   export function run(args: string[]): Promise<void>
 }
+declare module 'prettier/internal/legacy-cli.mjs' {
+  export function run(args: string[]): Promise<void>
+}

--- a/tests/integration/__tests__/cli.test.ts
+++ b/tests/integration/__tests__/cli.test.ts
@@ -148,8 +148,6 @@ describe('prettier-pnp cli', () => {
     const args = [
       '--pn',
       'organize-imports@~3.0.2',
-      '--pn',
-      'curly-and-jsdoc@0.118.0',
       'index.js',
     ]
 
@@ -242,7 +240,9 @@ describe('prettier-pnp cli', () => {
 
       const a = 5;
 
-      if (false) true;
+      if (false) {
+        true;
+      }
       "
     `)
   })
@@ -314,16 +314,7 @@ describe('prettier-pnp cli', () => {
       expect(result.status).toEqual(1)
       expect(result.stdout).toEqual('')
       expect(result.stderr.split('\n').slice(0, 8).join('\n'))
-        .toMatchInlineSnapshot(`
-        "npm ERR! code E404
-        npm ERR! 404 Not Found - GET https://registry.npmjs.org/prettier-plugin-curly_ - Not found
-        npm ERR! 404 
-        npm ERR! 404  'prettier-plugin-curly_@*' is not in this registry.
-        npm ERR! 404 
-        npm ERR! 404 Note that you can also install from a
-        npm ERR! 404 tarball, folder, http url, or git url.
-        "
-      `)
+        .toContain(`404 Not Found - GET https://registry.npmjs.org/prettier-plugin-curly_ - Not found`)
     })
   })
 })

--- a/tests/integration/__tests__/cli.test.ts
+++ b/tests/integration/__tests__/cli.test.ts
@@ -23,6 +23,7 @@ export function runPrettierPnpCli(...args: string[]) {
     {
       cwd: sandboxPath,
       env: process.env,
+      shell: true,
       encoding: 'utf-8',
     },
   )
@@ -47,6 +48,7 @@ describe('prettier-pnp cli', () => {
       ['install', '--no-package-lock'],
       {
         cwd: sandboxPath,
+        shell: true,
         stdio: 'inherit',
       },
     )
@@ -313,8 +315,7 @@ describe('prettier-pnp cli', () => {
       const result = runPrettierPnpCli(...args)
       expect(result.status).toEqual(1)
       expect(result.stdout).toEqual('')
-      expect(result.stderr.split('\n').slice(0, 8).join('\n'))
-        .toContain(`404 Not Found - GET https://registry.npmjs.org/prettier-plugin-curly_ - Not found`)
+      expect(result.stderr).toContain(`404 Not Found - GET https://registry.npmjs.org/prettier-plugin-curly_ - Not found`)
     })
   })
 })

